netconf_client/ncclient.py

```diff
@@ -32,9 +32,17 @@
 def _pretty_xml(xml):
     """Reformats a given string containing an XML document (for human readable output)"""
 
-    parser = etree.XMLParser(remove_blank_text=True)
-    tree = etree.fromstring(xml, parser)
-    return etree.tostring(tree, pretty_print=True).decode()
+    pretty = ""
+    try:
+        parser = etree.XMLParser(remove_blank_text=True)
+        tree = etree.fromstring(xml, parser)
+        pretty = etree.tostring(tree, pretty_print=True).decode()
+    except Exception as e:
```

Comment:
This needs to catch the specific type of exception

Reply:
Yes, good point. Would it be sufficient to catch the lxml.etree **base class** [Error](https://lxml.de/api/lxml.etree.Error-class.html)? And any other exception types (not derived directly or indirectly from Error) remain unhandled?

Reply:
Yes I think that would work

Reply:
Yes, it works fine. The unit-tests run successfully with this change. I re-submit...